src/Grpc/JsonTranscoding/test/Microsoft.AspNetCore.Grpc.JsonTranscoding.Tests/JsonTranscodingServiceMethodProviderTests.cs

```diff
@@ -79,6 +79,42 @@ public void AddMethod_NoHttpRuleInProto_ThrowNotFoundError()
         Assert.Equal("Couldn't find gRPC endpoint for method NoOption.", ex.Message);
     }
 
+    [Fact]
+    public void AddMethod_Unary_NoHttpRuleInProto_IncludeUnannotatedMethods_ResolveMethod()
+    {
+        // Arrange & Act
+        var endpoints = MapEndpoints<JsonTranscodingGreeterService>(configureJsonTranscoding: o =>
+        {
+            o.IncludeUnannotatedMethods = true;
+        });
+
+        // Assert
+        var endpoint = FindGrpcEndpoint(endpoints, nameof(JsonTranscodingGreeterService.NoOption));
+
+        Assert.Equal("/transcoding.JsonTranscodingGreeter/NoOption", endpoint.RoutePattern.RawText);
+        Assert.Equal("POST", endpoint.Metadata.GetMetadata<IHttpMethodMetadata>()?.HttpMethods.Single());
+        Assert.Equal("transcoding.JsonTranscodingGreeter.NoOption", endpoint.Metadata.GetMetadata<GrpcJsonTranscodingMetadata>()?.MethodDescriptor.FullName);
+        Assert.Null(endpoint.Metadata.GetMetadata<GrpcJsonTranscodingMetadata>()?.HttpRule);
```

Comment:
Is it valid to test a scenario where `IncludeUnnannotatedMethods` is true and there is an HttpRule?